src/components/container/OverviewTable/OverviewTable.tsx

```diff
@@ -0,0 +1,146 @@
+    const layoutContext = useContext(LayoutContext);
+
+    const [primaryData, setPrimaryData] = useState<OverviewData<PrimaryOverviewDataType>>();
+    const [secondaryData, setSecondaryData] = useState<OverviewData<SecondaryOverviewDataType>[]>();
+
+    const primaryHeaderBackgroundColor = resolveColor(layoutContext.colorScheme, props.primaryHeaderBackgroundColor ?? { lightMode: '#e2f1ff', darkMode: '#000' });
+    const primaryHeaderTextColor = resolveColor(layoutContext.colorScheme, props.primaryHeaderTextColor ?? { lightMode: '#000', darkMode: '#8bcbff' });
+
+    const goodValueBackgroundColor = resolveColor(layoutContext.colorScheme, props.goodValueBackgroundColor ?? { lightMode: '#c8ffc2', darkMode: '#000' });
+    const goodValueTextColor = resolveColor(layoutContext.colorScheme, props.goodValueTextColor ?? { lightMode: '#000', darkMode: '#96ff85' });
+
+    const notGoodValueBackgroundColor = resolveColor(layoutContext.colorScheme, props.notGoodValueBackgroundColor ?? { lightMode: '#ffedc2', darkMode: '#000' });
+    const notGoodValueTextColor = resolveColor(layoutContext.colorScheme, props.notGoodValueTextColor ?? { lightMode: '#000', darkMode: '#ffc847' });
+
+    const daysToCompute = props.daysToCompute ?? 28;
+    const today = startOfToday();
+    const startDate = add(today, { days: -daysToCompute });
+
+    const getDataProvider = <T extends OverviewDataType>(dataType: T): OverviewDataProvider<T> => {
+        if (isSurveyDataType(dataType.rawDataType)) {
+            return { type: dataType, dataProvider: getSurveyDataProvider(dataType.rawDataType) };
+        }
+        return { type: dataType, dataProvider: getDailyDataTypeDefinition(dataType.rawDataType).dataProvider };
+    };
+
+    const getPrimaryDataProvider = (): OverviewDataProvider<PrimaryOverviewDataType> => {
+        if (props.preview) {
+            return createPreviewDataProvider(props.primaryDataType);
+        }
+        return getDataProvider(props.primaryDataType);
+    };
+
+    const getSecondaryDataProviders = (): OverviewDataProvider<SecondaryOverviewDataType>[] => {
+        if (props.preview) {
+            return props.secondaryDataTypes.map(createPreviewDataProvider);
+        }
+        return props.secondaryDataTypes.map(getDataProvider);
+    };
+
+    useInitializeView(() => {
+        const primaryDataProvider = getPrimaryDataProvider();
+        primaryDataProvider.dataProvider(startDate, today).then(result => {
+            return { type: primaryDataProvider.type, queryResult: result };
+        }).then(setPrimaryData);
+
+        const secondaryDataProviders = getSecondaryDataProviders();
+        Promise.all(secondaryDataProviders.map(secondaryDataProvider => secondaryDataProvider.dataProvider(startDate, today).then(result => {
+            return { type: secondaryDataProvider.type, queryResult: result };
+        }))).then(setSecondaryData);
+    }, [], [props.preview, props.primaryDataType, props.secondaryDataTypes]);
```

Comment:
Along with my confusion about whether or not the Primary data shows up in the table... if you *do* want it to show up in the table, it looks like we'll query the exact same data twice?

Reply:
Yes, we would.

I think in practice it will be rare to have the same data type in both the primary and secondary configuration.

It wouldn't provide much value to see that when my steps are categorized as high (primary column), they are also high numerically (secondary column).  Likewise for other data types.

I think it might be a tad complex to try to share data providers across the primary/secondary boundary, so I'm not sure the juice is worth the squeeze given the expected usage of this component.